File: frontend/apps/android/app/src/main/java/app/pantopus/android/ui/screens/profile/PublicProfileSampleData.kt
@file:Suppress("PackageNaming", "MagicNumber")

package app.pantopus.android.ui.screens.profile

import app.pantopus.android.ui.screens.shared.content_detail.bodies.ProfileReviewCard
import app.pantopus.android.ui.theme.PantopusColors
import app.pantopus.android.ui.theme.PantopusIcon

/**
 * B.2 (A10.5) — deterministic neighbor-profile fixtures for the two
 * design frames (Derek R. populated · Sasha M. new neighbor). Backend has
 * been removed, so previews + Paparazzi project these directly; the
 * view-model synthesises the same shape from the live DTO.
 */
object PublicProfileSampleData {
    /** FRAME 1 — populated handyman with 47 reviews + 4 verifications. */
    val derekPopulated =
        NeighborProfileContent(
            hero =
                NeighborHero(
                    name = "Derek Reyes",
                    locality = "Elm Park · 5th & Elm",
                    avatarUrl = null,
                    isVerified = true,
                    identity = NeighborIdentity.Personal,
                    kicker = "Neighbor since 2022",
                ),
            stats =
                listOf(
                    NeighborStat(
                        id = "rating",
                        value = "4.9",
                        label = "47 reviews",
                        icon = PantopusIcon.Star,
                        iconColor = PantopusColors.warning,
                    ),
                    NeighborStat(id = "jobs", value = "32", label = "Jobs done"),
                    NeighborStat(id = "response", value = "~45m", label = "Response"),
                ),
            bio =
                "Handyman who lives two doors down. Hangs shelves, patches drywall, fixes the " +
                    "stuff your landlord won't. Bike-commutes everywhere — happy to drop a tool by " +
                    "on the way home. Weekend mornings work best.",
            skills = listOf("Handyman", "Patch & paint", "Plumbing (light)", "Cycling", "Help moving", "EN · ES"),
            verifications =
                listOf(
                    NeighborVerification(
                        id = "address",
                        icon = PantopusIcon.Home,
                        label = "Address",
                        meta = "Verified Mar 2022 · postcard",
                    ),
                    NeighborVerification(id = "identity", icon = PantopusIcon.BadgeCheck, label = "Identity", meta = "Government ID"),
                    NeighborVerification(id = "email", icon = PantopusIcon.Mail, label = "Email", meta = "derek@…"),
                    NeighborVerification(id = "phone", icon = PantopusIcon.Phone, label = "Phone", meta = "+1 ••• ••• 0193"),
                ),
            reviews =
                listOf(
                    ProfileReviewCard(
                        id = "r1",
                        reviewerName = "Maria K.",
                        reviewerAvatarUrl = null,
                        rating = 5,
                        body =
                            "Showed up early, brought his own anchors, and noticed the stud finder was " +
                                "lying. Shelves are level a month later. Already booked him for the closet.",
                        timestamp = "2w · for “Hang 3 shelves”",
                    ),
                ),
            reviewCount = 47,
            posts =
                listOf(
                    PublicProfilePost(
                        id = "dp1",
                        body = "Free leftover deck screws + a half-bag of anchors on my stoop. First come.",
                        timeAgo = "3d ago",
                        locality = "5th & Elm",
                        reactions = 9,
                        replies = 4,
                        intent = PublicProfilePost.Intent.Offer,
                    ),
                ),
            isNewNeighbor = false,
            primaryCtaLabel = "Message",
        )

    /** FRAME 2 — new verified neighbor, 0 reviews, graceful degradation. */
    val sashaNewNeighbor =
        NeighborProfileContent(
            hero =
                NeighborHero(
                    name = "Sasha Mendel",
                    locality = "Elm Park · 6th St",
                    avatarUrl = null,
                    isVerified = true,
                    identity = NeighborIdentity.Fresh,
                    kicker = "Joined 4 days ago",
                ),
            stats =
                listOf(
                    NeighborStat(
                        id = "rating",
                        value = "—",
                        label = "No reviews yet",
                        icon = PantopusIcon.Star,
                        valueColor = PantopusColors.appTextMuted,
                        iconColor = PantopusColors.appTextMuted,
                    ),
                    NeighborStat(id = "jobs", value = "0", label = "Jobs done"),
                    NeighborStat(id = "response", value = "New", label = "Response", valueColor = PantopusColors.primary600),
                ),
            bio = null,
            skills = emptyList(),
            verifications =
                listOf(
                    NeighborVerification(
                        id = "address",
                        icon = PantopusIcon.Home,
                        label = "Address verified",
                        meta = "Postcard delivered to 6th St",
                        tile = NeighborVerification.Tile.Success,
                        trailing = NeighborVerification.Trailing.Status("Today"),
                    ),
                    NeighborVerification(
                        id = "identity",
                        icon = PantopusIcon.BadgeCheck,
                        label = "Identity verified",
                        meta = "Government ID matched",
                        tile = NeighborVerification.Tile.Success,
                        trailing = NeighborVerification.Trailing.Status("4d ago"),
                    ),
                    NeighborVerification(
                        id = "email",
                        icon = PantopusIcon.Mail,
                        label = "Email confirmed",
                        meta = "sasha@…",
                        tile = NeighborVerification.Tile.Success,
                        trailing = NeighborVerification.Trailing.Status("4d ago"),
                    ),
                ),
            reviews = emptyList(),
            reviewCount = 0,
            mutuals =
                NeighborMutuals(
                    count = 4,
                    names = "Jamal, Ravi, Lena, Amina",
                    initials = listOf("JT", "RD", "LP", "AS"),
                ),
            welcome =
                NeighborWelcome(
                    title = "Be the welcome wagon",
                    body =
                        "Sasha just moved in. A quick “hi, welcome to the block” goes a long way — " +
                            "and first messages from verified neighbors travel fast.",
                ),
            isNewNeighbor = true,
            primaryCtaLabel = "Say hi",
        )
}

File: frontend/apps/ios/Pantopus/Features/Profile/PublicProfileSampleData.swift
//
//  PublicProfileSampleData.swift
//  Pantopus
//
//  B.2 (A10.5) — deterministic neighbor-profile fixtures for the two
//  design frames (Derek R. populated · Sasha M. new neighbor). Backend
//  has been removed, so previews + tests project these directly; the
//  view-model synthesises the same shape from the live DTO.
//

import Foundation

public enum PublicProfileSampleData {
    /// FRAME 1 — populated handyman with 47 reviews + 4 verifications.
    public static let derekPopulated = NeighborProfileContent(
        hero: NeighborHero(
            name: "Derek Reyes",
            locality: "Elm Park · 5th & Elm",
            avatarURL: nil,
            isVerified: true,
            identity: .personal,
            kicker: "Neighbor since 2022"
        ),
        stats: [
            NeighborStat(
                id: "rating",
                value: "4.9",
                label: "47 reviews",
                icon: .star,
                iconColor: Theme.Color.warning
            ),
            NeighborStat(id: "jobs", value: "32", label: "Jobs done"),
            NeighborStat(id: "response", value: "~45m", label: "Response")
        ],
        bio: "Handyman who lives two doors down. Hangs shelves, patches drywall, fixes the "
            + "stuff your landlord won't. Bike-commutes everywhere — happy to drop a tool by "
            + "on the way home. Weekend mornings work best.",
        skills: ["Handyman", "Patch & paint", "Plumbing (light)", "Cycling", "Help moving", "EN · ES"],
        verifications: [
            NeighborVerification(id: "address", icon: .home, label: "Address", meta: "Verified Mar 2022 · postcard"),
            NeighborVerification(id: "identity", icon: .badgeCheck, label: "Identity", meta: "Government ID"),
            NeighborVerification(id: "email", icon: .mail, label: "Email", meta: "derek@…"),
            NeighborVerification(id: "phone", icon: .phone, label: "Phone", meta: "+1 ••• ••• 0193")
        ],
        reviews: [
            ProfileReviewCard(
                id: "r1",
                reviewerName: "Maria K.",
                reviewerAvatarURL: nil,
                rating: 5,
                body: "Showed up early, brought his own anchors, and noticed the stud finder was "
                    + "lying. Shelves are level a month later. Already booked him for the closet.",
                timestamp: "2w · for \u{201C}Hang 3 shelves\u{201D}"
            )
        ],
        reviewCount: 47,
        posts: [
            PublicProfilePost(
                id: "dp1",
                body: "Free leftover deck screws + a half-bag of anchors on my stoop. First come.",
                timeAgo: "3d ago",
                locality: "5th & Elm",
                reactions: 9,
                replies: 4,
                intent: .offer
            )
        ],
        isNewNeighbor: false,
        primaryCtaLabel: "Message"
    )

    /// FRAME 2 — new verified neighbor, 0 reviews, graceful degradation.
    public static let sashaNewNeighbor = NeighborProfileContent(
        hero: NeighborHero(
            name: "Sasha Mendel",
            locality: "Elm Park · 6th St",
            avatarURL: nil,
            isVerified: true,
            identity: .fresh,
            kicker: "Joined 4 days ago"
        ),
        stats: [
            NeighborStat(
                id: "rating",
                value: "—",
                label: "No reviews yet",
                icon: .star,
                valueColor: Theme.Color.appTextMuted,
                iconColor: Theme.Color.appTextMuted
            ),
            NeighborStat(id: "jobs", value: "0", label: "Jobs done"),
            NeighborStat(id: "response", value: "New", label: "Response", valueColor: Theme.Color.primary600)
        ],
        bio: nil,
        skills: [],
        verifications: [
            NeighborVerification(
                id: "address",
                icon: .home,
                label: "Address verified",
                meta: "Postcard delivered to 6th St",
                tile: .success,
                trailing: .status("Today")
            ),
            NeighborVerification(
                id: "identity",
                icon: .badgeCheck,
                label: "Identity verified",
                meta: "Government ID matched",
                tile: .success,
                trailing: .status("4d ago")
            ),
            NeighborVerification(
                id: "email",
                icon: .mail,
                label: "Email confirmed",
                meta: "sasha@…",
                tile: .success,
                trailing: .status("4d ago")
            )
        ],
        reviews: [],
        reviewCount: 0,
        mutuals: NeighborMutuals(
            count: 4,
            names: "Jamal, Ravi, Lena, Amina",
            initials: ["JT", "RD", "LP", "AS"]
        ),
        welcome: NeighborWelcome(
            title: "Be the welcome wagon",
            body: "Sasha just moved in. A quick \u{201C}hi, welcome to the block\u{201D} goes a long way — "
                + "and first messages from verified neighbors travel fast."
        ),
        posts: [],
        isNewNeighbor: true,
        primaryCtaLabel: "Say hi"
    )
}
